src/readqtccalls.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * Tlf - contest logging program for amateur radio operators
+ * Copyright (C) 2013 Ervin Hegedüs - HA2OS <airween@gmail.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+ */
+	/* ------------------------------------------------------------
+	 *        Read sent QTC QSO's
+	 *
+	 *--------------------------------------------------------------*/
+
+#include "readqtccalls.h"
+#include "get_time.h"
+#include "globalvars.h"
+#include "tlf.h"
+#include "qtcutil.h"
+#include <glib.h>
+
+#include <curses.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <time.h>
+#include <ctype.h>
+
+extern int qtcdirection;
+extern GHashTable* qtc_store;
+extern struct t_qtc_store_obj *qtc_empty_obj;
+extern int nr_qsosflags_for_qtc;
+extern int nr_qsos;
+
+int readqtccalls()
+{
+    int s = 0;
+    char inputbuffer[160];
+    FILE *fp;
+    char temps[30], callsign[15];
+    int tempi;
+    int last_qtc = 0;
+    int i;
+
+    clear();
+
+    if (qtc_store != NULL) {
+	g_hash_table_destroy(qtc_store);
+    }
+    if (qtc_empty_obj != NULL) {
+	g_free(qtc_empty_obj);
+    }
+    qtc_store = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, NULL);
+    qtc_empty_obj = g_malloc0(sizeof (struct t_qtc_store_obj));
+    qtc_empty_obj->total = 0;
+    qtc_empty_obj->received = 0;
+    qtc_empty_obj->sent = 0;
+    nr_qsosflags_for_qtc = nr_qsos;
```

Comment:
I would suggest to put line 56..67 into a init_qtc_store() function or similar in qtcutil.c. Then we can move the qtc_store definition from main.c into qtcutil.c too. No one else needs to know about the hash table.
The same goes for qtc_empty_obj then.
What do you think?